### ISSUE TITLE
can't use bookmarks with write-file and helm-ff-run-copy-file

### ISSUE THREAD

**Issue by CeleritasCelery** [emacs-helm/helm#2088]
## behavior
When calling `write-file` helm opens a prompt to select the destination. However if you use bookmarks to save common locations (as recommended in #2073) and then try and use a bookmark to change locations, it will change your session type from `write file` to `helm-find-files`. This means that when you select a destination file it will open it instead of writing the file. The same thing happens when trying to use helms built-in  `helm-ff-run-copy-file`. Basically you can't use bookmarks with operation that are not file navigation. 

## Steps to reproduce (recipe)
open a `helm-find-files` session
select file and run `helm-ff-run-copy-file`
use a bookmark to change directories.
observe that the session type has changed.
attempt to copy the file.
observe that the destination is opened instead of the file copied to it.

## Describe versions of Helm, Emacs, operating system, etc.
latest helm, Emacs 26.1

## Are you using `emacs-helm.sh` to reproduce this bug:
no

**Comment by alphapapa** [comment]
> Are you using emacs-helm.sh to reproduce this bug:
> no

Why not?  :)

**Comment by thierryvolpiatto** [comment]
Yes bookmarks are just for `helm-find-files` and their action is to jump to the bookmark, so yes what you are asking for is indeed not working.
For copying files use the HFF history to jump quickly to an already visited directory.

**Comment by thierryvolpiatto** [comment]
Note also that you may run a modified version of helm because normally helm prevent running bookmarks in the both situation you described an error is sent.

helm: Error: Trying to run helm within a running helm session

**Comment by thierryvolpiatto** [comment]
You can try `(add-to-list 'helm-completing-read-handlers-alist '(write-file . helm-read-file-name-handler-1))`
to have HFF history, same for `basic-save-buffer`, etc...
